cmd/launcher/launcher.go

```diff
@@ -54,8 +56,33 @@ const (
 // enabled, the finalizers will trigger various restarts.
 func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) error {
 	logger := log.With(ctxlog.FromContext(ctx), "caller", log.DefaultCaller)
+
+	// If delay_start is configured, wait before running launcher.
+	if opts.DelayStart > 0*time.Second {
+		level.Debug(logger).Log(
+			"msg", "delay_start configured, waiting before starting launcher",
+			"delay_start", opts.DelayStart.String(),
+		)
+		time.Sleep(opts.DelayStart)
+	}
+
 	level.Debug(logger).Log("msg", "runLauncher starting")
 
+	// We've seen launcher intermittently be unable to recover from
+	// DNS failures in the past, so this check gives us a little bit
+	// of room to ensure that we are able to resolve DNS requests
+	// before proceeding with starting launcher.
+	if err := backoff.WaitFor(func() error {
+		_, lookupErr := net.LookupIP(opts.KolideServerURL)
+		return lookupErr
+	}, 10*time.Second, 1*time.Second); err != nil {
+		level.Info(logger).Log(
+			"msg", "could not successfully perform IP lookup before starting launcher, proceeding anyway",
```

Comment:
Hrm. I was going to suggest a longer timeout before exiting, but if you're going to proceed anyhow, I guess this is reasonable

Reply:
In test, mine was resolving within 3-5 seconds fairly consistently, so I think this is safe 🤞

Reply:
I am also happy to bump it up, though, I don't feel very strongly about it being 10 sec if you have a timeout you prefer!

Reply:
Nah, it seems fine to try.